plugins/idea/pluginBase/src/main/java/com/imaginea/kodebeagle/base/settings/ui/SearchPanel.java

```diff
@@ -0,0 +1,140 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.imaginea.kodebeagle.base.settings.ui;
+
+import com.imaginea.kodebeagle.base.action.RefreshActionBase;
+import com.imaginea.kodebeagle.base.model.Search;
+import com.intellij.openapi.ui.ComboBox;
+
+import javax.swing.*;
+import javax.swing.border.TitledBorder;
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+
+public class SearchPanel {
+
+    private static final String TITLE4 = "KodeBeagle Api Server";
+    private static final String KODEBEAGLE_SEARCH_API_URL = "KodeBeagle Api URL:";
+    private static final String OVERRIDE = "Override";
+
+    private static final int[] KODEBEAGLE_SEARCH_PANEL_COLUMN_WIDTHS =
+            new int[] {29, 84, 63, 285, 0, 95, 0};
+
+    private static final int[] KODEBEAGLE_SEARCH_PANEL_ROW_HEIGHTS = new int[] {0, 0, 0};
+
+    private static final double[] KODEBEAGLE_SEARCH_PANEL_COLUMN_WEIGHTS =
+            new double[] {0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 1.0E-4};
+
+    private static final double[] KODEBEAGLE_SEARCH_PANEL_ROW_WEIGHTS =
+            new double[] {0.0, 0.0, 1.0E-4};
+
+    private final GridBagConstraints kodebeagleSearchPanelVerticalSpacer1 =
+            new GridBagConstraints(0, 0, 1, 1, 0.0, 0.0, GridBagConstraints.CENTER,
+                    GridBagConstraints.BOTH, new Insets(0, 0, 5, 5), 0, 0);
+
+    private final GridBagConstraints kodebeagleSearchPanelFirstLeft =
+            new GridBagConstraints(1, 1, 5, 1, 0.0, 0.0, GridBagConstraints.CENTER,
+                    GridBagConstraints.BOTH, new Insets(0, 0, 0, 0), 0, 0);
+
+    private final GridBagConstraints kodebeagleSearchPanelFirstCenter =
+            new GridBagConstraints(3, 1, 2, 1, 0.0, 0.0, GridBagConstraints.CENTER,
+                    GridBagConstraints.BOTH, new Insets(0, 0, 0, 5), 0, 0);
+
+    private final GridBagConstraints kodebeagleSearchPanelFirstRight =
+            new GridBagConstraints(5, 1, 1, 1, 0.0, 0.0, GridBagConstraints.CENTER,
+                    GridBagConstraints.BOTH, new Insets(0, 0, 0, 0), 0, 0);
+
+    private final JPanel spacer = new JPanel(null);
+    private ComboBox kbApiURLComboBox;
+    private JCheckBox kbApiURLOverrideCheckBox;
+
+    protected SearchPanel() {
+        createFields();
+    }
+
+    public final ComboBox getKbApiURLComboBox() {
+        return kbApiURLComboBox;
+    }
+
+    public final JCheckBox getKbApiURLOverrideCheckBox() {
+        return kbApiURLOverrideCheckBox;
+    }
+
+    private void createFields() {
+        kbApiURLComboBox = new ComboBox();
+        kbApiURLComboBox.setEditable(true);
+        kbApiURLComboBox.setVisible(true);
+        kbApiURLOverrideCheckBox = new JCheckBox();
+        kbApiURLOverrideCheckBox.setVisible(true);
+        kbApiURLOverrideCheckBox.setText(OVERRIDE);
+        kbApiURLOverrideCheckBox.addActionListener(new ActionListener() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Make this anonymous inner class a lambda (sonar.java.source not set. Assuming 8 or greater.) [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1604)